### PR TITLE
style: update chart line and grid styles and colors

### LINKED ISSUE #2459: [Grids] Minor style update
In order to reduce visual abundance on timeseries, I would suggest to slightly change a couple of style details of our grids.

1. **Avoid dashed grid lines**
_This normally happens on horizontal grid lines_

2. **Avoid vertical grid lines with darker color**
_We use this to visually highlight relevant moments in time, but the result is that the chart look more confusing and too rich_

3. **Check grid lines color**
_Light theme: #E9EDF3
Dark theme: #343741_

4. **Bucket dashed line on hover color: _EUI.colors.darkShade_**

The following images show the expected final result that we should get with these changes:

Screenshot 2024-06-07 at 15 25 29

Screenshot 2024-06-07 at 15 25 55

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

Updates line and color styles.

Changes:
- [x] Remove dashed lines for horizontal gridlines
- [x] Set `LIGHT_THEME` grid lines color to `#E9EDF3`

No Changes required (as is currently):
- Set `DARK_THEME` grid lines color to `#343741` (i.e. `EUI.colors.lightShade`)
- Bucket dashed line on hover color (i.e. `crosshair.line`) set to `EUI.colors.darkShade`

## Issues

closes #2459

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated
- [x] Visual changes have been tested with `light` and `dark` themes

### CODE REVIEW

File: packages/charts/src/utils/themes/dark_theme.ts
Comment (nickofthyme) [inline packages/charts/src/utils/themes/dark_theme.ts:1]: @gvnmagni Can you take a look at the requested changes?

> **Avoid vertical grid lines with darker color**
> We use this to visually highlight relevant moments in time, but the result is that the chart look more confusing and too rich

I have concerns about this request. I believe you are talking about the gridlines on the multi-layer time axis here. These are actually rendered with progressively darker colors but are all the same width.

That said, the darker colors are used to indicate an increased unit of time, take the example below where a the lightest line would be a week, the slightly darker a month, and the darkest a quarter year.

image

If we do away with this darkening the time intervals become ambiguous and the tick/gridlines are confusing.

image

Comment (nickofthyme) [inline packages/charts/src/utils/themes/dark_theme.ts:1]: As for you comment about...

> **Avoid vertical grid lines with darker color**
> We use this to visually highlight relevant moments in time, but the result is that the chart look more confusing and too rich

I have concerns about this request. I believe you are talking about the gridlines on the multi-layer time axis here. These are actually rendered with progressively darker colors but are all the same width.

That said, the darker colors are used to indicate an increased unit of time, take the example below where a the lightest line would be a week, the slightly darker a month, and the darkest a quarter year.

image

If we do away with this darkening the time intervals become ambiguous and the tick/gridlines are confusing.

image

**Review by nickofthyme — COMMENTED** [review]
@gvnmagni Can you take a look at the requested changes? Here is a [link](https://ech-e2e-ci--pr-2492-975mz1g8.web.app/storybook/?path=/story/axes--tick-label-rotation) to the storybook with the gridline options.

File: packages/charts/src/utils/themes/dark_theme.ts
Comment (gvnmagni) [reply packages/charts/src/utils/themes/dark_theme.ts:1]: This is a valid point but to be honest I prefer the new option, which I find way less intense in terms of visual look&feel, the first option is really messy. But this is a completely open discussion that I would love to have with you all

One additional point, it seems to me (please correct me if I am wrong) that in most cases that Monthly label matches smaller ticks and this problem gets vastly resized

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Comment by nickofthyme** [comment]
Large VRT diff (900+) due to changes in gridline `dash` and color styles.

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Comment by nickofthyme** [comment]
buildkite test this

**Review by mbondyra — APPROVED** [review]

**Pull request merged**